src/query/api/v1/handler/prometheus/common.go

```diff
@@ -152,6 +156,61 @@ func parseTagCompletionQuery(r *http.Request) (string, error) {
 	return queries[0], nil
 }
 
+func parseTimeWithDefault(
+	r *http.Request,
+	key string,
+	defaultTime time.Time,
+) (time.Time, error) {
+	if t := r.FormValue(key); t != "" {
+		return util.ParseTimeString(t)
+	}
+
+	return defaultTime, nil
+}
+
+// ParseSeriesMatchQuery parses all params from the GET request
+func ParseSeriesMatchQuery(
+	r *http.Request,
+	tagOptions models.TagOptions,
+) (*storage.SeriesMatchQuery, *xhttp.ParseError) {
+	r.ParseForm()
+	matcherValues := r.Form["match[]"]
+	if len(matcherValues) == 0 {
+		return nil, xhttp.NewParseError(errors.ErrInvalidMatchers, http.StatusBadRequest)
+	}
+
+	start, err := parseTimeWithDefault(r, "start", time.Time{})
```

Comment:
Prometheus does from timeMin to timeMax so this should mirror that. We shouldn't get issues with a long time range since we don't fetch any points

Potentially could be an issue if we have old IDs in a glacier cluster, but even then it shouldn't be too bad if the match query is well constructed

Reply:
Changed it to a 40 day lookback after conferring offline